src/core/continuous_space.jl

```diff
@@ -246,15 +246,19 @@ export nearest_neighbor, elastic_collision!, interacting_pairs
 """
     nearest_neighbor(agent, model, r) → nearest
 Return the agent that has the closest distance to given `agent`, according to the
-space's metric.
-Valid only in continuous space.
+space's metric. Valid only in continuous space.
+Return `nothing` if no agent is within distance `r`.
 """
 function nearest_neighbor(agent, model, r)
   n = space_neighbors(agent, model, r)
   length(n) == 0 && return nothing
   d, j = Inf, 0
   for i in 1:length(n)
-    @inbounds dnew = sqrt(sum(abs2.(agent.pos .- model[n[i]].pos)))
+    if model.space.metric == "euclidean"

```

Comment:
Any reason `metric` is a string? Would it not be better as a `Symbol` or abstract type?

Reply:
Yes symbol is better. I am debating whether it is worth it for us to add `Distances` as a dependency and have the metric be `<: Metric`...

Reply:
But unfortunately `Distances` doesn't support Tuples, which is what we use here, so this idea is out of the window.